package/MDAnalysis/transformations/translate.py

```diff
@@ -138,4 +150,173 @@ def wrapped(ts):
+    except AttributeError:
+        if center_of == 'mass':
+            raise AttributeError('{} is not an AtomGroup object with masses'.format(ag))
+        else:
+            raise ValueError('{} is not an AtomGroup object'.format(ag))
+
+    def wrapped(ts):
+        boxcenter = ts.triclinic_dimensions.sum(axis=0) / 2.0
+        position = boxcenter
+        if center_to != 'center':
+            _origin = center_to
+            position[plane] = _origin[plane]
+        vector = np.asarray(position, np.float32) - center_method()
+        ts.positions += vector
+        
+        return ts
+    
+    return wrapped
+
+
+def center_in_axis(ag, axis, center_to="center", center_of='geometry', wrap=False):
+    """
+    Translates the coordinates of a given :class:`~MDAnalysis.coordinates.base.Timestep`
+    instance so that the center of geometry/mass of the given :class:`~MDAnalysis.core.groups.AtomGroup`
+    is centered on an axis. This axis can only be parallel to the x, y or z planes. 
+    
+    Example
+    -------
+    
+    Translate the center of mass of of the second residue of the universe u to the center of the line 
+    parallel to the x axis that passes through the [0, 0, 1] point:
+    
+    .. code-block:: python
+    
+        ag = u.residues[1].atoms
+        transform = MDAnalysis.transformations.center_in_axis(ag, 'x', [0,0,1], center_of='mass')
+        u.trajectory.add_transformation(transform)
+    
+    Parameters
+    ----------
+    ag: AtomGroup
+        atom group to be centered on the unit cell.
+    axis: str
+        used to define the plane on which the given AtomGroup will be centered. Defined as
+        a string or an array-like of the plane and the value of its translation. Suported 
+        planes are x, y and z.
+    center_to: array-like or str
+        coordinate on which the axes are centered. Can be an array of three coordinates or
+        `center` to shift the origin to the center of the unit cell. Default is `None` 
+        which sets [0, 0, 0] as the origin of the axis. 
+    center_of: str, optional
+        used to choose the method of centering on the given atom group. Can be 'geometry'
+        or 'mass'
+    wrap: bool, optional
+        If `True`, all the atoms from the given AtomGroup will be moved to the unit cell
+        before calculating the center of mass or geometry. Default is `False`, no changes
+        to the atom coordinates are done before calculating the center of the AtomGroup. 
+    
+    Returns
+    -------
+    MDAnalysis.coordinates.base.Timestep
+    
+    """
+    pbc_arg = wrap
+    if axis is not None:
+        if axis not in ('x', 'y', 'z'):
+            raise ValueError('{} is not a valid axis'.format(axis))
+        axes = {'x' : 0, 'y': 1, 'z' : 2}
+        axis = axes[axis]
+    if center_to is not None:
+        if isinstance(center_to, string_types):
+            if not center_to == 'center':
```

Comment:
unnecessary check and nesting of if clauses. You can leave out the `isinstance` check in the first place. Here you also have an unnecessary nesting of the if clause. It would be better to write the two tests into the same if-clause.